src/mesh/impls/bout/boutmesh.cxx

```diff
@@ -839,6 +839,26 @@ int BoutMesh::load() {
   // Add boundary regions
   addBoundaryRegions();
 
+  // Create CELL_CENTRE Coordinates object
+  addCoordinates(CELL_CENTRE);
+
+  if (StaggerGrids) {
+    // Add Coordinates objects at staggered locations, if there are enough grid
+    // points.
+    // This is a temporary workaround. In v4.3 we will users to call
+    // REQUEST_LOCATION(location) for each location that is needed and remove
+    // this so that we don't waste memory on unneeded Coordinates.
+    if (LocalNx >= 4) {
+      addCoordinates(CELL_XLOW);
+    }
+    if (LocalNy >= 4) {
+      addCoordinates(CELL_YLOW);
+    }
+    // Can always add ZLOW Coordinates, since z-interpolation on Field2D is a
+    // null operation
+    addCoordinates(CELL_ZLOW);
+  }
```

Comment:
This logic is in `Coordinates::geometry` too -- does it need to be here?

Reply:
I think it does need to be here. This code ensures the `Coordinates` are created at each location initially. The code in `Coordinates::geometry()` replaces (and therefore re-calculates) the staggered `Coordinates` objects after the `CELL_CENTRE` `Coordinates` is changed by `geometry()`. The idea is that #1392 will update things so that here we only create the `CELL_CENTRE` `Coordinates` (i.e. lines 845-860 are deleted), the user has to call `Mesh::addCoordinates(location)` for any needed staggered location, and if the user calls `geometry()` it has to be before the staggered `Coordinates` are initialized.

Maybe there is a neater way, but I don't think we want to call `Coordinates::geometry()` here because it would duplicate work already done when `geometryNoRecalculate` is called as part of the constructor in `addCoordinates(CELL_CENTRE)`.